src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -392,61 +367,51 @@ public void TrimExcess()
             int threshold = (int)(_nodes.Length * 0.9);
             if (_size < threshold)
             {
-                SetCapacity(_size);
+                Array.Resize(ref _nodes, _size);
+                _version++;
             }
         }
 
         /// <summary>
-        /// Ensures the queue has enough space to add another item.
+        /// Ensures that the priority queue has the specified capacity
+        /// and resizes its underlying array if necessary.
         /// </summary>
-        private void EnsureEnoughCapacityBeforeAddingNode()
+        private void EnsureCapacityCore(int capacity)
```

Comment:
This is far from inlineable.  Considering it's called before every add, it might be worth adding a streamlined:
```C#
if (_nodes.Length < capacity)
    Grow();
```
as presumably then it would be inlineable.

Reply:
I did consider that approach but ultimately reused what exists in [`List<T>`](https://github.com/dotnet/runtime/blob/e7204f5d6fcaca5e097ec854b3be6055229fc442/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L399).

Reply:
But the code you linked to in `List<T>` isn't the whole story.  For individual Adds, which in PriorityQueue is used even for ranges, `List<T>` only hits the code path you cited after it's already done its own inline check:
https://github.com/dotnet/runtime/blob/e7204f5d6fcaca5e097ec854b3be6055229fc442/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L195-L220

Reply:
I updated the code so that checks are moved outside of the method (kept as an assert in the core Grow method).